packages/beeq/src/components/progress/bq-progress.tsx

```diff
@@ -0,0 +1,184 @@
+import { Component, Element, h, Prop, Watch } from '@stencil/core';
+
+import {
+  PROGRESS_MODE,
+  PROGRESS_THICKNESS,
+  PROGRESS_TYPE,
+  TProgressMode,
+  TProgressThickness,
+  TProgressType,
+} from './bq-progress.types';
+import { validatePropValue } from '../../shared/utils';
+
+@Component({
+  tag: 'bq-progress',
+  styleUrl: './scss/bq-progress.scss',
+  shadow: true,
+})
+export class BqProgress {
+  // Own Properties
+  // ====================
+
+  private previousValue: number | null = null;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqProgressElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** It defines the mode of progress bar to display */
+  @Prop({ reflect: true }) mode: TProgressMode = 'determinate';
```

Comment:
I would rather recommend to rename this property to `indeterminate` as a boolean (default `false`) and when `indeterminate` is set, we could just:

```jsx
<progress class={progressClasses} value={this.indeterminate ? undefined : this.value} max="100" />
...
{this.enableTooltip && !this.indeterminate && ( ... )}
...
<div
  style={{ fontVariant: 'tabular-nums' }}
  class={{
    ' font-medium tabular-nums leading-regular text-text-primary': true,
    'text-ui-danger': this.type === 'error',
    hidden: !this.label || this.indeterminate,
  }}
>
  {this.value}%
</div>
```